contrib/gorm.io/gorm.v1/gorm_test.go

```diff
@@ -215,15 +216,26 @@ func TestCallbacks(t *testing.T) {
 		a.Equal(ext.SpanTypeSQL, span.Tag(ext.SpanType))
 		a.Equal(queryText, span.Tag(ext.ResourceName))
 		a.Equal("gorm.io/gorm.v1", span.Tag(ext.Component))
+		a.Equal(parentSpan.Context().SpanID(), span.ParentID())
+
+		for _, s := range spans {
+			if s.Tag(ext.Component) == "jackc/pgx.v5" {
+				// The underlying driver should receive the gorm span
+				a.Equal(span.SpanID(), s.ParentID())
+			}
+		}
```

Comment:
```suggestion
		spans := mt.FinishedSpans()
		require.Len(t, spans, 5)
		
		{
			parentSpan := spans[4]
			gormSpan := spans[3]
			pgx1 := spans[2]
			pgx2 := spans[1]
			pgx3 := spans[0]

			assert.Equal(t, "gorm.create", gormSpan.OperationName())
			assert.Equal(t, ext.SpanTypeSQL, gormSpan.Tag(ext.SpanType))
			assert.Equal(t, queryText, gormSpan.Tag(ext.ResourceName))
			assert.Equal(t, "gorm.io/gorm.v1", gormSpan.Tag(ext.Component))

			assert.Equal(t, gormSpan.ParentID(), parentSpan.SpanID())
			assert.Equal(t, pgx1.ParentID(), gormSpan.SpanID())
			assert.Equal(t, pgx2.ParentID(), gormSpan.SpanID())
			assert.Equal(t, pgx3.ParentID(), gormSpan.SpanID())
		}
```

I think I would prefer the test to be a little bit "dumber". Now that the `mocktracer.Start()` lines are added to each specific subtest, we can assert a concrete number of spans, and the expected behavior is more explicit.

(feel free to ignore this refactor if you prefer so, I can do it myself in a subsequent PR)